src/Haskell/Ide/Engine/Plugin/Base.hs

```diff
@@ -105,11 +106,11 @@ version =
 hieGhcDisplayVersion :: String
 hieGhcDisplayVersion = compilerName ++ "-" ++ VERSION_ghc
 
-getProjectGhcVersion :: IO String
-getProjectGhcVersion = do
+getProjectGhcVersion :: BIOS.Cradle -> IO String
+getProjectGhcVersion crdl = do
   isStackProject   <- doesFileExist "stack.yaml"
   isStackInstalled <- isJust <$> findExecutable "stack"
-  if isStackProject && isStackInstalled
+  if BIOS.actionName (BIOS.cradleOptsProg crdl) == "stack" && isStackProject && isStackInstalled
```

Comment:
This is dirty. We should upstream this function, same as for `ModuleCache` which defines the same function